File: 백준/Silver/17074. 정렬/정렬.py
import sys

input = sys.stdin.readline

n = int(input())  # 100000
arr = list(map(int,input().split()))

cnt = 0
tmp = 0

for i in range(n - 1):
    if arr[i] > arr[i + 1]:
        tmp = i + 1
        cnt+=1

#print(tmp)

if cnt >= 2:
    print(0)
    exit()
elif cnt == 0:
    print(n)
else:
    ans = 0
    if tmp == 1 or arr[tmp] >= arr[tmp - 2]:
        ans += 1
    if tmp == n - 1 or arr[tmp + 1] >= arr[tmp - 1]:
        ans += 1
    print(ans)

File: 백준/Silver/17074. 정렬/정렬.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        StringTokenizer st;
        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int tmp = 0;
        int cnt = 0;
        for (int i = 0; i < n - 1; i++) {
            if(arr[i] > arr[i + 1]){
                cnt++;
                tmp = i + 1;
            }
        }

        if(cnt == 0){
            System.out.println(n);
        }else if(cnt >= 2){
            System.out.println(0);
        }else{
            int ans = 0;
            if(tmp == 1 || arr[tmp] >= arr[tmp - 2]){
                ans++;
            }
            if(tmp == n - 1 || arr[tmp + 1] >= arr[tmp - 1]){
                ans++;
            }
            System.out.println(ans);
        }
    }
}
